reference/format/datagrid.rst

```diff
@@ -0,0 +1,166 @@
+
+    ``template`` (**type**: ``string``)
+
+        This option is only available when the ``frontend_type`` is ``html`` and the value of the
+        ``type`` option is ``twig``. Its value is a template reference.
+
+    ``type`` (**type**: ``string``)
+
+        When the ``frontend_type`` is ``string``, this option can be set to ``translatable`` to
+        provide translated values.
+
+        When it is set to ``twig``, the Twig template referenced to with the ``template`` option
+        will be rendered.
+
+``extends`` (**type**: ``string``)
+
+    You can reuse existing datagrid configurations by passing their name here.
+
+``filters`` (**type**: ``map``)
+
+    This option currently only provides the ``columns`` key which maps column names to a map which
+    configures how a particular column can be filtered. Available options to configure how to
+    filter a column are:
+
+    ``data_name`` (**type**: ``string``)
+
+        The name of the column from the data source whose values will be filtered.
+
+    ``type`` (**type**: ``string``)
+
+        The type of the filter to be used in the UI.
+
+``options`` (**type**: ``map``)
+
+    Available additional options are:
+
+    ``base_datagrid_class`` (**type**: ``string`` **default**: ``Oro\Bundle\DatagridBundle\Datagrid\Datagrid``)
+
+        With this option, you can switch the datagrid class to a custom implementation.
+
+    ``export`` (**type**: ``boolean`` **default**: ``false``)
+
+        When enabled, the user can export the datagrid in CSV format.
+
+``properties`` (**type**: ``map``)
+
+    This is used for two things: configure how to determine the id of each row and configure links
+    that will be reused in the actions.
+
+    ``id`` (**type**: ``string`` **default**: ``~``)
+
+        The name of the property that acts as an identifier for each entry. By default, the ``id``
+        property is assumed.
+
+    To configure links, use a unique string as an identifier and pass it a map with the following
+    options:
+
+    ``callable`` (**type**: ``string``)
+
+        An expression that will be evaluated when the link is generated.
+
+    ``params`` (**type**: ``map``)
+
+        Additional parameters that are passed to the URL generator together with the configured
+        route name.
+
+    ``route`` (**type**: ``string``)
+
+        The name of the route to the controller action that should be called.
+
+    ``type`` (**type**: ``string``)
+
+        Can be either ``route`` or ``callable`` to use a statically configured route or to
+        dynamically generate a link.
+
+``sorters`` (**type**: ``map``)
+
+    The only available key is ``columns`` which is the map that contains an entry for each column
+    the user can sort the grid by. Each key is the name of a column and the value is a map with
+    the key ``data_name`` mapped to the data source column that will be used to sort the grid.
+
+``source`` (**type**: ``map``)
```

Comment:
You forgot for the main sub-parameter of `sourece` parameter is `query`.
`query` parameter allow to build the dql query for given datagrid.